protected.go

```diff
@@ -180,7 +190,7 @@ func (p *Protector) dial(op ops.Op, network, addr string, timeout time.Duration)
 	}
 	copy(conn.ip[:], ip.To4())
 
-	socketFd, err := syscall.Socket(syscall.AF_INET, syscall.SOCK_STREAM, 0)
+	socketFd, err := syscall.Socket(syscall.AF_INET, socketType, 0)
```

Comment:
Actually, now that I think about it, we've effectively not been distinguishing on that before and haven't had any issues with it, so maybe we just leave it alone?